presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergTableLayoutHandle.java

```diff
@@ -13,31 +13,101 @@
  */
 package com.facebook.presto.iceberg;
 
+import com.facebook.presto.common.Subfield;
 import com.facebook.presto.common.predicate.TupleDomain;
 import com.facebook.presto.spi.ColumnHandle;
 import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.relation.RowExpression;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
 
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
 public class IcebergTableLayoutHandle
         implements ConnectorTableLayoutHandle
 {
+    private final List<IcebergColumnHandle> partitionColumns;
+    private final boolean pushdownFilterEnabled;
+    private final TupleDomain<Subfield> domainPredicate;
+    private final RowExpression remainingPredicate;
+    private final Map<String, IcebergColumnHandle> predicateColumns;
+    private final TupleDomain<ColumnHandle> partitionColumnPredicate;
+    private final Optional<Set<IcebergColumnHandle>> requestedColumns;
     private final IcebergTableHandle table;
     private final TupleDomain<ColumnHandle> tupleDomain;
```

Comment:
We need to either remove the original tupleDomain or subclass. This is used for Iceberg to filter files. Can you check how Hive does it when filter pushdown was set to false(In HiveMetadata.java ln 2742)? One possible way is to reconstruct it using the new fields when filtering the files. We just need to make sure the transformation is equivalent.

Reply:
@yingsu00 Yes we can remove the original tupleDomain. It can be constructed by transforming domainPredicate field.